src/shared/wizards/multiStepWizard.ts

```diff
@@ -12,17 +12,74 @@ import * as vscode from 'vscode'
 import * as picker from '../../shared/ui/picker'
 
 export interface WizardStep {
-    (): Thenable<WizardStep | undefined>
+    (): Thenable<Transition>
+}
+
+export interface Transition {
+    nextState: NextWizardState
+    nextStep?: WizardStep
+}
+
+export enum NextWizardState {
```

Comment:
Consider adding comments to these enums so that reader does not have to go to the switch statement to infer their meaning.